paladin/mod.ts

```diff
@@ -0,0 +1,203 @@
+import { Drash } from "../deps.ts";
+
+type ReferrerPolicy =
+  | ""
+  | "no-referrer"
+  | "no-referrer-when-downgrade"
+  | "same-origin"
+  | "origin"
+  | "strict-origin"
+  | "origin-when-cross-origin"
+  | "strict-origin-when-cross-origin"
+  | "unsafe-url";
+
+interface Configs {
+  "X-XSS-Protection"?: boolean;
+  "Referrer-Policy"?: ReferrerPolicy;
+  "X-Content-Type-Options"?: boolean;
+  hsts?: {
+    max_age?: boolean | number;
+    include_sub_domains?: boolean;
+    preload?: boolean;
+  };
+  "X-Powered-By"?: boolean | string;
+  "X-Frame-Options"?: "DENY" | "SAMEORIGIN" | boolean | string // eg ALLOW-FROM www.example.com
+  ;
+  expect_ct?: {
+    enforce?: boolean;
+    max_age?: number;
+    report_uri?: string;
+  };
+  "X-DNS-Prefetch-Control"?: boolean;
+  "Content-Security-Policy"?: string;
+}
+
+/**
+ * A logger middleware inspired by https://www.npmjs.com/package/morgan.
+ *
+ * @param configs - See IDexterConfigs
+ */
+export function Paladin(
+  configs: Configs = {},
+) {
+
+  if (!configs.hsts) {
+    configs.hsts = {};
+  }
+  if (!configs.expect_ct) {
+    configs.expect_ct = {};
+  }
+
+  // Default configs when no `configs` param is passed in
+  const defaultConfigs = {
+    "X-XSS-Protection": "1; mode=block",
+    "X-Content-Type-Options": "nosniff",
+    hsts: {
+      max_age: 5184000, // 60 days
+      include_sub_domains: "include_sub_domains",
+    },
+    "X-Powered-By": false, // False for removing the header
+    "X-Frame-Options": "SAMEORIGIN",
+    "X-DNS-Prefetch-Control": false,
+  };
+
+  /**
+   * The middleware function that's called by Drash.
+   *
+   * @param request - The request object.
+   * @param response - (optional) The response object.
+   */
+  function paladin(
+    request: Drash.Http.Request,
+    response?: Drash.Http.Response,
+  ): void {
+    // If there is a response, then we know this is occurring after the request
+    if (response) {
+      // Set "X-XSS-Protection" header. See https://helmetjs.github.io/docs/xss-filter/
+      if (configs["X-XSS-Protection"] !== false) {
+        response.headers.set(
+          "X-XSS-Protection",
+          defaultConfigs["X-XSS-Protection"],
+        );
+        configs["X-XSS-Protection"] = true;
+      }
+
+      // Set "Referrer-Policy" header if passed in. See https://helmetjs.github.io/docs/referrer-policy/
+      if (configs["Referrer-Policy"]) {
```

Comment:
this is a large function if we can map these if statements to functions instead like
```
func assignXss(response) {
        response.headers.set(
          "X-XSS-Protection",
          defaultConfigs["X-XSS-Protection"],
        );
        configs["X-XSS-Protection"] = true;
}

funcs: {
  ""X-XSS-Protection": assignXss...
}

if (configs["X-XSS-Protection"]) funcs["X-XSS-Protection"](response)
```

whatever it may be i just think it'd be more readable

Reply:
I'll give it a try 👍

Reply:
how about this:
![image](https://user-images.githubusercontent.com/47337480/87822018-b3fbd080-c868-11ea-91fe-66623b263cab.png)

Reply:
let's move this to an improvement issue. we can reiterate on this there.

Reply:
Alright, makes sense due to the discussions - though should be easy to do anyways

Reply:
https://github.com/drashland/deno-drash-middleware/issues/27